### PR TITLE
Rename Realm.App.getCachedApp to Realm.App.getApp

### PR DESCRIPTION
**Pull request by kneth**
## ☑️ ToDos

* [x] 📝 Changelog entry
* ~[ ] 📝 `Compatibility` label is updated or copied from previous entry~
* ~[ ] 🚦 Tests~
* [x] 📝 Public documentation PR created or is not necessary
* [x] 💥 `Breaking` label has been applied or is not necessary

*If this PR adds or changes public API's:*
* [x] typescript definitions file is updated
* [x] jsdoc files updated
* ~[ ] Chrome debug API is updated if API is available on React Native~

### CODE REVIEW

**Comment by kraenhansen** [comment]
Do we have any tests exercising this?
I don't think this method have [the same semantics as in Realm Web](https://github.com/realm/realm-js/blob/v10/packages/realm-web/src/index.ts#L29-L37).

What happens if a user calls `getApp` before instantiating an app elsewhere with that particular id?

```javascript
import { App } from "realm";
const app = App.getApp("my-awesome-app");
```

**Comment by kneth** [comment]
> Do we have any tests exercising this?

Not at the moment.

> Ex: What happens if a user calls `getApp` before instantiating an app elsewhere with that particular id?

`null` will be returned. I agree that it will be valuable to create the app if it hasn't been created.

**Comment by kneth** [comment]
@kraenhansen I have changed the behaviour to mimic Realm Web.

File: lib/app.js [hunk 1]
```diff
@@ -19,6 +19,7 @@
 "use strict";
 
 const {promisify} = require("./utils.js");
+const { realpath } = require("fs");
```
Comment (kraenhansen) [inline lib/app.js:22]: This looks like an auto-complete error? :)

**Pull request merged**